### ISSUE TITLE
Reauthentication error

### ISSUE THREAD

**Issue by larena1** [custom-components/alexa_media_player#1512]
**Describe the bug**
Reauthentication is impossible. Opening the Amazon website from the addon will show an error.
It seems to be related to the cookiejar conversion between aiohttp and httpx.
Only way to get it going again is removing the integration and setting it up again.

**To Reproduce**
1. Reauthenticate

**Expected behavior**

**Screenshots**

**System details**
 - Home-assistant (version): latest
 - Hassio (Yes/No): (Please note you may have to restart hassio 2-3 times to load the latest version of alexapy after an update. This looks like a HA bug).
 - alexa_media  (version from `const.py` or HA startup): 3.10.15
 - alexapy (version from `pip show alexapy` or HA startup): 1.25.3 (authcaptureproxy 1.1.1)
 - Amazon [2FA is enabled](https://github.com/custom-components/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account) (y/n). We will not debug login issues if unanswered: yes

**Logs**
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 220, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 79, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 219, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 137, in handle
    result = await result
  File "/config/custom_components/alexa_media/config_flow.py", line 1122, in wrapped
    return await cls.handler(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/authcaptureproxy/auth_capture_proxy.py", line 387, in all_handler
    resp = await getattr(self.session, method)(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1736, in get
    return await self.request(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1500, in request
    request = self.build_request(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 351, in build_request
    return Request(
  File "/usr/local/lib/python3.9/site-packages/httpx/_models.py", line 1099, in __init__
    Cookies(cookies).set_cookie_header(self)
  File "/usr/local/lib/python3.9/site-packages/httpx/_models.py", line 1792, in set_cookie_header
    self.jar.add_cookie_header(urllib_request)
  File "/usr/local/lib/python3.9/http/cookiejar.py", line 1367, in add_cookie_header
    attrs = self._cookie_attrs(cookies)
  File "/usr/local/lib/python3.9/http/cookiejar.py", line 1326, in _cookie_attrs
    self.non_word_re.search(cookie.value) and version > 0):
TypeError: expected string or bytes-like object
```
**Additional context**

**Comment by alandtse** [comment]
"Latest" doesn't tell me anything. Try again.

**Comment by larena1** [comment]
Sorry, I assumed you'd be aware of the latest versions of the components. Have added the versions for you.

**Comment by larena1** [comment]
What happens is apparently `filter_cookies` returns `SimpleCookie` or `BaseCookie` while `httpx` expects a `CookieJar`

https://docs.aiohttp.org/en/v3.8.1/_modules/aiohttp/cookiejar.html

Did this work for you when you implemented it in https://github.com/custom-components/alexa_media_player/commit/598175f3e8297aea256a22417565df79638c1d8b?

**Comment by alandtse** [comment]
> Did this work for you when you implemented it in [598175f](https://github.com/custom-components/alexa_media_player/commit/598175f3e8297aea256a22417565df79638c1d8b)?

Yes.

> What happens is apparently `filter_cookies` returns `SimpleCookie` or `BaseCookie` while `httpx` expects a `CookieJar`

Where are you seeing a call to filter_cookies?

**Comment by larena1** [comment]
> > Did this work for you when you implemented it in [598175f](https://github.com/custom-components/alexa_media_player/commit/598175f3e8297aea256a22417565df79638c1d8b)?
> 
> Yes.

Is it possible that back then you only tried to authenticate but not to reauthenticate? Because only then there will be cookies passed between aiohttp and httpx.

> > What happens is apparently `filter_cookies` returns `SimpleCookie` or `BaseCookie` while `httpx` expects a `CookieJar`
> 
> Where are you seeing a call to filter_cookies?

Right where you implemented it? https://github.com/custom-components/alexa_media_player/commit/598175f3e8297aea256a22417565df79638c1d8b#diff-46d832f19ab1f0952ac20eb0d870c53343adcea7636327a6dc16889f0704aecfL384

I was able to fix the initial issue like follows

```
            cookies = self.login._session.cookie_jar.filter_cookies(
                self.proxy._host_url.with_path("/")
            )
            cookies = {k: v.value for k, v in cookies.items()}
            self.proxy.session.cookies = cookies
```

It was then possible to log in through the auth capture proxy but after that another issue came up

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 220, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 79, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 219, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 137, in handle
    result = await result
  File "/config/custom_components/alexa_media/config_flow.py", line 1127, in wrapped
    return await cls.handler(request, **kwargs)
  File "/config/custom_components/alexa_media/lib/authcaptureproxy/auth_capture_proxy.py", line 404, in all_handler
    result = await run_func(test, test_name, resp, self.data, self.query)
  File "/config/custom_components/alexa_media/lib/authcaptureproxy/helper.py", line 123, in run_func
    result = await func(*args, **kwargs)
  File "/config/custom_components/alexa_media/lib/alexapy/alexaproxy.py", line 65, in test_amazon_url
    self._login.session.cookie_jar.update_cookies(self.session.cookies)
  File "/config/custom_components/alexa_media/lib/alexapy/aiohttp/cookiejar.py", line 167, in update_cookies
    for name, cookie in cookies:
  File "/usr/local/lib/python3.9/_collections_abc.py", line 851, in __iter__
    yield (key, self._mapping[key])
  File "/usr/local/lib/python3.9/site-packages/httpx/_models.py", line 1882, in __getitem__
    value = self.get(name)
  File "/usr/local/lib/python3.9/site-packages/httpx/_models.py", line 1834, in get
    raise CookieConflict(message)
httpx.CookieConflict: Multiple cookies exist with name=session-id
```

**Comment by alandtse** [comment]
I'm just going to remove the resume session path. It's too much effort to deal with due to the incompatibilities between aiohttp and httpx.
I also don't know how you have any aiohttp cookies to pass on to httpx. They were deprecated a year ago and should've been removed once you removed and readded the component.

**Comment by larena1** [comment]
https://github.com/custom-components/alexa_media_player/blob/dev/custom_components/alexa_media/config_flow.py#L383

Not sure what you mean by deprecated? All I'm doing is trying to reauthenticate via the auth capture proxy when the integration asks me for it.

**Comment by alandtse** [comment]
aiohttp isn't used by the component and hasn't been used for at least a year. There's some legacy code from before the switch, but by now, no one should be having aiohttp cookies anymore.

Remove and readd the component. That is what the reauth will be moving forward.

**Comment by larena1** [comment]
> Remove and readd the component. That is what the reauth will be moving forward.

Exactly what I'm doing for the past months every few weeks but that's more than inconvenient. I think the reason there being cookies in the aiohttp cookie jar might then be related to the second issue that came up when I fixed the first. That will call update_cookies on the aiohttp cookie jar with httpx' cookies after auth via the proxy.

**Comment by alandtse** [comment]
Delete any pickle files in the custom component directory.

**Comment by alandtse** [comment]
Or use the force logout option so it deletes the pickle file for you. Then relogin. Your symptoms  look like you're not getting an oauth token and you're using an outdated cookie. Proper logs would've shown this but you didn't provide any.

**Comment by larena1** [comment]
I always deleted the pickle file when I had to readd the integration and there's an oauth token set in the config entry and oauth_login is also true.

Which symptoms exactly? The issue when forced to reauth?

**Comment by alandtse** [comment]
Reauth every few weeks; presence of a aiohttp cookie a year after the component stopped using it.

**Comment by larena1** [comment]
https://gitlab.com/keatontaylor/alexapy/-/blob/dev/alexapy/alexaproxy.py#L65

Cookie jar of aiohttp is populated right there and then persisted with pickle. Doesn't everybody have aiohttp cookies then when using the auth capture proxy?

**Comment by larena1** [comment]
I'll see what I can do. So the integration does solely rely on the retrieved oauth token now and does not need the cookies and pickle file?

Are you not using the integration yourself anymore or just not interested in maintaining it further?

**Comment by alandtse** [comment]
> I'll see what I can do. So the integration does solely rely on the retrieved oauth token now and does not need the cookies and pickle file?
> 
> Are you not using the integration yourself anymore or just not interested in maintaining it further? 

It still uses the pickle to save the cookies.

I use it but I've never had the auth problems. I just am spending my time on other interests. Again, if I had time, I would've removed the reauth path and removed legacy login options. They generate too many repeat issues.

**Comment by Marnalas** [comment]
In case it helps someone.

I had the exact same problem you describe @larena1 (exact same logs). I tried what you proposed and had, again, the exact same result.
```
            cookies = self.login._session.cookie_jar.filter_cookies(
                self.proxy._host_url.with_path("/")
            )
            cookies = {k: v.value for k, v in cookies.items()}
            self.proxy.session.cookies = cookies
```

While investigating I found out that I had a remnant of a precious version in my configuration.yaml I had completly forgotten about : 
```
alexa_media:
  accounts:
    - email: !secret amazon_user
      password: !secret amazon_password
      url: amazon.com
```

I then did this:
1. I removed this configuration ;
2. I restarted HomeAssistant ;
3. I removed the HACS integration ;
4. I added the HACS integration again (3.10.15) ;
5. I restarted HomeAssistant ;
6. I added the integration and configured it following the standard way (didn't touch any checkbox).
It then worked on the first try.

@alandtse thank you for your work. I enjoy it a great deal.
